service/history/historyEngine.go

```diff
@@ -2425,3 +2430,40 @@ func (e *historyEngineImpl) GetReplicationMessages(ctx ctx.Context, taskID int64
 	e.logger.Debug("Successfully fetched replication messages.", tag.Counter(len(replicationMessages.ReplicationTasks)))
 	return replicationMessages, nil
 }
+
+func (e *historyEngineImpl) ReapplyEvents(
+	ctx ctx.Context,
+	reapplyRequest *h.ReapplyEventsRequest,
+) error {
+
+	domainEntry, err := e.getActiveDomainEntry(reapplyRequest.DomainID)
+	if err != nil {
+		return err
+	}
+	domainID := domainEntry.GetInfo().ID
+	execution := *reapplyRequest.GetWorkflowExecution()
```

Comment:
reapplication should target at current workflow, so why execution? (meaning that the run ID inside can be set)

Reply:
remove the run id here to only apply to current run